src/profiling_data.rs

```diff
@@ -1,17 +1,73 @@
 use crate::event::Event;
+use crate::{ProfilerFiles, RawEvent, StringTable};
+use std::fs;
+use std::mem;
 use std::path::Path;
+use std::time::{Duration, SystemTime};
 
-pub struct ProfilingData {}
+pub struct ProfilingData {
+    event_data: Vec<u8>,
+    string_table: StringTable,
+}
 
 impl ProfilingData {
-    pub fn new(_path_stem: &Path) -> ProfilingData {
-        unimplemented!()
+    pub fn new(path_stem: &Path) -> ProfilingData {
+        let paths = ProfilerFiles::new(path_stem);
+
+        let string_data = fs::read(paths.string_data_file).expect("couldn't read string_data file");
+        let index_data = fs::read(paths.string_index_file).expect("couldn't read string_index file");
+        let event_data = fs::read(paths.events_file).expect("couldn't read events file");
+
+        let string_table = StringTable::new(string_data, index_data);
+
+        ProfilingData {
+            string_table,
+            event_data,
+        }
     }
 
-    pub fn iter_events<'a, F>(&'a self, mut _f: F)
-    where
-        F: FnMut(&Event<'a>),
-    {
-        unimplemented!()
+    pub fn iter(&self) -> impl Iterator<Item = Event<'_>> {
```

Comment:
One advantage of the callback based approach is that memory allocations for strings and stuff can be re-used, which is not possible with an iterator (I think).

Reply:
I believe that was the reason I made the change you commented on [here](https://github.com/rust-lang/measureme/pull/5#discussion_r268630912)